crates/partition-store/src/invocation_status_table/mod.rs

```diff
@@ -212,6 +220,84 @@ impl ScanInvocationStatusTable for PartitionStore {
         )
         .map_err(|_| StorageError::OperationalError)
     }
+
+    fn for_each_invocation_status<
+        F: FnMut((InvocationId, InvocationStatus)) -> ControlFlow<()> + Send + Sync + 'static,
+    >(
+        &self,
+        range: RangeInclusive<PartitionKey>,
+        f: F,
+    ) -> Result<impl Future<Output = Result<()>> + Send> {
+        // these two iterators can run concurrently in theory.
```

Comment:
this is unpleasant. opinions welcome on what to do here. i dont think the iteratos can move across threads during their iteration, but nonetheless the functions are expected to be Send so i can only use a tokio mutex guard here as those are Send. I could use parking lot with the send_guard feature enabled, but thats not something i want to enable for all our usages, I think. I could also not use a guard and lock on every row.